### PR TITLE
3rd inode refactor

### PR DESCRIPTION
**Pull request by Ngoguey42**
This is a simplification for `inode_intf`.

To be used as a basis for #1249 and #1251

### CODE REVIEW

File: src/irmin-pack/inode.ml [hunk 1]
```diff
@@ -776,41 +783,37 @@ module Make_ext
...
 
   let mem t k = Pack.mem t k
-
-  let unsafe_find ~check_integrity t k =
-    match Pack.unsafe_find ~check_integrity t k with
-    | None -> None
-    | Some v ->
-        let v = Inter.Val_impl.of_bin v in
-        Some v
+  let unsafe_find ~check_integrity t k = Pack.unsafe_find ~check_integrity t k
```
Comment (icristescu) [inline src/irmin-pack/inode.ml:797]: if the only usage of `unsafe_find` is in `find` maybe you can move this directly there?

**Review by icristescu — APPROVED** [review]
Thanks!

**Comment by Ngoguey42** [comment]
I will try to merge this one this afternoon if no one else has any comments.

File: src/irmin-pack/inode_intf.ml [hunk 2]
```diff
@@ -65,41 +61,31 @@ module type INTER = sig
...
 
-module type S_EXT = sig
-  include Irmin.CONTENT_ADDRESSABLE_STORE
-
-  val batch : [ `Read ] t -> ([ `Read | `Write ] t -> 'a Lwt.t) -> 'a Lwt.t
+    (* The following symbols are only available from codes that directly call
+       [Make_inter], e.g. [Make_ext], inode_layers.ml and test_inode.ml
+       The other pieces of code will only see [Val] as a [Val_intf], through
+       [S].
+    *)
```
Comment (CraigFe) [inline src/irmin-pack/inode_intf.ml:74]: I think this is either going to be (a) obvious from the code or (b) made incorrect on a future revision. Perhaps just replace it with a doc-comment saying that this is an unstable internal API?

**Review by CraigFe — APPROVED** [review]
LGTM, thanks.

**Pull request merged**